src/utilities/MegaProfile.js

```diff
@@ -0,0 +1,68 @@
+import axios from 'axios';
+import { ApiGatewayAuth } from './ApiGatewayAuth';
+
+export class MegaProfile {
+  constructor(config = {}) {
+    // Set variables
+    const host = config.host || process.env.MEGAPROFILE_HOST;
+    const path = config.path || process.env.MEGAPROFILE_PATH;
+    const builtUrl = `${host}${path}`;
+    this.url = config.url || process.env.MEGAPROFILE_URL || builtUrl;
+    this.auth = config.auth || new ApiGatewayAuth();
+
+    // Create Client
+    this.client = axios.create({
+      baseURL: this.url,
+      // Other request defaults? Headers?
+      params: {
+        token: this.authenticate(),
+      },
```

Comment:
```suggestion
      params: {
        token: this.auth.token.access_token,
      },
```

Might as well authenticate and fetch that token prior to setting the param or remove this whole part and just call the authenticate function directly here at the end of the constructor.

Reply:
Remove this as it `this.authenticate()` never returns anything and you are setting the object value directly later.